### ISSUE TITLE
23.9 regression: Package not found when using pins

### ISSUE THREAD

**Issue by jonashaag** [conda/conda-libmamba-solver#327]
### Checklist

- [X] I added a descriptive title
- [X] I searched open reports and couldn't find a duplicate

### What happened?

`conda install` seems to not find packages if there's a pin for them. This is a regression from 23.7 to 23.9

```
$ conda create -p /tmp/env -y
$ echo 'xtensor=0.22' > /tmp/env/conda-meta/pinned

$ conda install -p /tmp/env 'xtensor>0' --solver=libmamba
...
PackagesNotFoundError: The following packages are not available from current channels:

  - xtensor[version='>0']


$ conda install -p /tmp/env 'xtensor' --solver=libmamba
OK
```

**Comment by jaimergp** [comment]
I'm afk but I wonder if `>0` returns any results for xtensor. Is it interpreted as `>=1`.

**Comment by jonashaag** [comment]
Maybe bad example but it doesn't matter what constraint you are giving. It always fails.

**Comment by jaimergp** [comment]
It only fails with _some_ specs (those that contain `<` or `>`), and is interestingly caused by the `libmambapy.Query` spec parsing.

```
$ micromamba repoquery search "ca-certificates[version='<2024']" -c conda-forge
# finds nothing
micromamba repoquery search "ca-certificates<2024" -c conda-forge
 Name            Version     Build      Channel              
──────────────────────────────────────────────────────────────
 ca-certificates 2023.7.22   hf0a4a13_0 conda-forge/osx-arm64
 ca-certificates 2023.5.7    hf0a4a13_0 conda-forge/osx-arm64
 ca-certificates 2022.12.7   h4653dfc_0 conda-forge/osx-arm64
 ca-certificates 2022.9.24   h4653dfc_0 conda-forge/osx-arm64
 ca-certificates 2022.9.14   h4653dfc_0 conda-forge/osx-arm64
 ca-certificates 2022.6.15.2 h4653dfc_0 conda-forge/osx-arm64
 ca-certificates 2022.6.15.1 h4653dfc_0 conda-forge/osx-arm64
 ca-certificates 2022.6.15   h4653dfc_0 conda-forge/osx-arm64
 ca-certificates 2022.5.18.1 h4653dfc_0 conda-forge/osx-arm64
 ca-certificates 2022.5.18   h4653dfc_0 conda-forge/osx-arm64
 ca-certificates 2021.10.8   h4653dfc_0 conda-forge/osx-arm64
 ca-certificates 2021.5.30   h4653dfc_0 conda-forge/osx-arm64
 ca-certificates 2020.12.5   h4653dfc_0 conda-forge/osx-arm64
 ca-certificates 2020.11.8   h4653dfc_0 conda-forge/osx-arm64
 ca-certificates 2020.6.20   hecda079_0 conda-forge/osx-arm64
```

I am using `str(MatchSpec("cli-spec"))` to search the index with `libmambapy.Query` for compatible records with the pin. Since `xtensor>0` gets roundtripped as `xtensor[version='>0']` and this syntax is not supported by `Query`, it finds nothing and raises `PackagesNotFoundError`.

We can workaround this by using `MatchSpec.conda_build_form()`, but it we will lose the channel info in the process, so maybe we can use `channel::name=version=build` instead? Are there any docs for which syntax is supported by `Query`? Thanks!

**Comment by jaimergp** [comment]
See https://github.com/conda/conda-libmamba-solver/pull/328

**Comment by jaimergp** [comment]
This is now fixed in 23.9.3. Thanks for the report @jonashaag!
